### PR TITLE
Add support for cross-compiling via the Makefile

### PR DESCRIPTION
**Pull request by athre0z**
This adds support for cross-compiling in `all`, `lint`, and `test` Makefile targets.

Stuff that should work now, regardless of host architecture, assuming that QEMU+binfmt rules and gcc cross compiler is installed (see Dockerfile changes):

```
# build agent
make TARGET_ARCH=arm64
make clean # required!
make TARGET_ARCH=amd64

# run tests
make test TARGET_ARCH=arm64
make clean # required!
make test TARGET_ARCH=amd64
```

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -2,11 +2,18 @@ FROM debian:testing
 
 WORKDIR /agent
 
-ARG arch=amd64
```
Comment (athre0z) [inline Dockerfile:5]: I got rid of this: it doesn't appear to be used.

File: support/ebpf/Makefile [hunk 2]
```diff
@@ -1,28 +1,24 @@
...
 ifeq ($(NATIVE_ARCH),x86_64)
-NATIVE_ARCH:=x86
-else ifeq ($(NATIVE_ARCH),aarch64)
-NATIVE_ARCH:=arm64
+NATIVE_ARCH := amd64
+else ifneq (,$(filter $(NATIVE_ARCH),aarch64 arm64))
+NATIVE_ARCH := arm64
 else
 $(error Unsupported architecture: $(NATIVE_ARCH))
 endif
```
Comment (athre0z) [inline support/ebpf/Makefile:16]: Synchronized this code block with the equivalent block from main makefile.

File: support/ebpf/Makefile [hunk 3]
```diff
@@ -53,28 +49,28 @@ all: $(TRACER_NAME)
 debug: TARGET_FLAGS+=$(DEBUG_FLAGS)
 debug: all
 
-x86:
-	$(MAKE) target_arch=x86 all
+amd64:
+	$(MAKE) TARGET_ARCH=amd64 all
```
Comment (athre0z) [inline support/ebpf/Makefile:53]: Switched to consistently use the `GOARCH` style architecture names everywhere.

**Comment by rockdaboot** [comment]
Can you please add documentation in `README.md`?

I could cross-compile and run the arm64 binary on Debian (until it stops due to the expected error):
```
$ make clean
$ make TARGET_ARCH=arm64
$ file ./otel-profiling-agent
./otel-profiling-agent: ELF 64-bit LSB executable, ARM aarch64, version 1 (GNU/Linux), statically linked, BuildID[sha1]=5e41dd9a5aa2910a625192dc887886b02c90bdcf, for GNU/Linux 3.7.0, with debug_info, not stripped
$ sudo ./otel-profiling-agent
INFO[0000] Starting OTEL profiling agent v0.0.0 (revision joel/crosscompile-661a35ee, build timestamp 1720591948) 
INFO[0000] Automatically determining environment and machine ID ... 
INFO[0013] Environment: hardware, machine ID: 0xdbfd5da74c2d531b 
ERRO[0013] Failed to probe eBPF syscall: eBPF syscall is not available on your system
```

Beforehand, some packages needed to be installed:
```
sudo apt-get install gcc-aarch64-linux-gnu libc6-arm64-cross qemu-user qemu-user-static binfmt-support
```

`make test TARGET_ARCH=arm64` fails because the tests are compiled dynamically:
```
aarch64-binfmt-P: Could not open '/lib/ld-linux-aarch64.so.1': No such file or directory
```
To fix this, it likely requires
```
sudo dpkg --add-architecture amd64
sudo apt-get update
sudo apt-get upgrade
```
which I didn't test.

**Comment by rockdaboot** [comment]
`make agent TARGET_ARCH=arm64` generates an amd64 binary
```
make docker-image
make agent TARGET_ARCH=arm64
$ file otel-profiling-agent
otel-profiling-agent: ELF 64-bit LSB executable, x86-64, version 1 (GNU/Linux), statically linked, BuildID[sha1]=cab04d76401e9e880675cb161d4e7e34500188fd, for GNU/Linux 3.2.0, with debug_info, not stripped
```

**Comment by athre0z** [comment]
> `make test TARGET_ARCH=arm64` fails because the tests are compiled dynamically

You can get this working by running:

```
dpkg --add-architecture arm64
sudo apt update
sudo apt install libc6:arm64
```

> `make agent TARGET_ARCH=arm64` generates an amd64 binary

Hmm weird, I'll take a look.

**Pull request closed without merging**